<launch>
  <arg name="model" default="burger" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="move_forward_only" default="false"/>
  <arg name="configuration_basename" default="turtlebot3_lds_2d.lua"/>

  <!-- turtle bot declation --> 
  <arg name='second_jb' default="jb_1"/>
  <arg name="second_jb_x_pos" default=" 0.0"/>
  <arg name="second_jb_y_pos" default=" -8.0"/>
  <arg name="second_jb_z_pos" default=" 0.0"/>
  <arg name="second_jb_yaw"   default=" 0.0"/>
  
  <arg name='first_jb' default="jb_0"/>
  <arg name="first_jb_x_pos" default="-4.0"/>
  <arg name="first_jb_y_pos" default=" -8.0"/>
  <arg name="first_jb_z_pos" default=" 0.0"/>
  <arg name="first_jb_yaw"   default=" 0.0"/>


  

  <!-- launch gazebos from the data -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find multirobot_with_llm)/worlds/jetbot_house.world"/>
    <!-- <arg name="world_name" value="$(find ros_multi_jb)/worlds/turtlebot3_house_mod.world"/> -->
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>  

<!-- ################################################################################################# -->
  <group ns = "$(arg first_jb)">
    <!-- <param name="robot_description" command="$(find xacro)/xacro $(find jetbot_world)/urdf/jetbot_csi.urdf.xacro botname:=$(arg first_jb)" />   turtlebot3_burger.urdf -->
    <param name="robot_description" command="$(find xacro)/xacro $(find ros_multi_tb3)/urdf/turtlebot3_burger.urdf.xacro botname:=$(arg first_jb)" /> 
    
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
      <param name="publish_frequency" type="double" value="50.0" />
      <param name="tf_prefix" value="$(arg first_jb)" />
    </node>
    
    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model $(arg first_jb) -x $(arg first_jb_x_pos) -y $(arg first_jb_y_pos) -z $(arg first_jb_z_pos) -Y $(arg first_jb_yaw) -param robot_description" />


  </group>

<!-- ################################################################################################# -->
  <group ns = "$(arg second_jb)">
    <!-- <param name="robot_description" command="$(find xacro)/xacro $(find jetbot_world)/urdf/jetbot_csi.urdf.xacro botname:=$(arg second_jb)" /> -->
    <param name="robot_description" command="$(find xacro)/xacro $(find ros_multi_tb3)/urdf/turtlebot3_burger.urdf.xacro botname:=$(arg second_jb)" /> 
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
      <param name="publish_frequency" type="double" value="50.0" />
      <param name="tf_prefix" value="$(arg second_jb)" />
    </node>

    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model $(arg second_jb) -x $(arg second_jb_x_pos) -y $(arg second_jb_y_pos) -z $(arg second_jb_z_pos) -Y $(arg second_jb_yaw) -param robot_description" />

  </group>

</launch>
